server/main/templates/edd_base.html

```diff
@@ -76,12 +76,13 @@
             {% if user.is_authenticated %}
             <li class="dropdown" id="notification-dropdown">
               <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button"
-                  aria-haspopup="true" aria-expanded="false">
+                  aria-haspopup="menu" aria-controls="notifications-list" aria-expanded="false">
                 <span class="fas fa-inbox"></span>
                 <span class="badge"></span>
+                <span class="sr-only">Notifications</span>
               </a>
-              <ul class="dropdown-menu" id="notification-list">
-                <li class="message-empty">{% trans 'No new notifications.' %}</li>
+              <ul class="dropdown-menu" id="notification-list" role="menu">
+                <li role="none" class="message-empty"><a href="" aria-disabled="true" role="menuitem">{% trans 'No new notifications.' %}</a></li>
```

Comment:
I'm curious, what is the purpose of wrapping the list item here in an `<A>`? Should the dynamically added messages also be wrapped like this? How would it handle notifications that themselves contain links, in that case?

Reply:
Good question! Using `role="menu"` on the `ul` means that the children (marked with `role="menu item"` have to be "interactive", so buttons or links. This affects how screen readers allow users to navigate between items. Like you point out, that doesn't really work for our use case, since we have items that themselves contain links.

So ultimately instead of this approach we're going with a hybrid, similar to the second option described in (this article](https://www.smashingmagazine.com/2017/11/building-accessible-menu-systems/). It's not quite as satisfying in that doesn't provide quite as many affordances to assistive tech users, but since we aren't modeling a traditional pull-down menu, it's a better fit for what the app does.